desktop/src/main/java/bisq/desktop/main/offer/bsq_swap/create_offer/BsqSwapCreateOfferViewModel.java

```diff
@@ -572,6 +575,33 @@ private void updateButtonDisableState() {
         isPlaceOfferButtonDisabled.set(createOfferRequested || !inputDataValid || miningPoW.get());
     }
 
+    private void maybeInitializeWithData() {
```

Comment:
Maybe rename to `maybeApplyFromDuplicateOffer` or the like, to make the use case more clear. I check if the offer is null at the caller would also help to make it more clear when its applied.

Reply:
I kept the naming more general, as I though that there could be other use cases as well. Like some onboarding form in the future that automatic pre-fills a buy BSQ form.
Regarding the offer null check. Where do you want to see this?

Reply:
I meant to have the if check for null could be done at the caller of `maybeInitializeWithData` and then its not a maybe anymore... but no strong opinion either... It just took a bit more effort to understand the code that ways as one need to jump to the method to see whats really going on there to see that its only an optional case if the param is null.